tests/core/channels/test_slack.py

```diff
@@ -901,3 +915,64 @@ def test_slack_verify_signature_missing_headers():
     slack = SlackInput("mytoken", slack_signing_secret="foobar")
 
     assert slack.is_request_from_slack_authentic(request) is False
+
+
+async def fake_on_new_message(message: UserMessage):
+    pass
+
+
+@pytest.mark.asyncio
+async def test_slack_process_message_retry():
+    input_channel = SlackInput(
+        slack_token="YOUR_SLACK_TOKEN",
+        slack_channel="YOUR_SLACK_CHANNEL",
+        slack_signing_secret="foobar",
+    )
+
+    request = Mock()
```

Comment:
I have a suggestion here to use `responses.RequestsMock()` context manager instead of `Mock()`. Here is the [documentation link](https://github.com/getsentry/responses#responses-as-a-context-manager), you can pass the localhost URL and the headers to this object via `add` method. Then you can access the request from the context manager like so `request_mock.calls[0].request`. 

Have a try in a time-boxed effort but if it doesn't work out, keeping the tests as they are also works, just let me know. We use this context manager in a few other tests in `test_telemetry` and `test_duckling_entity_extractor` modules if you want to have a look.

Reply:
I am not sure if that can be used, because here I don't need to mock an api call like a `requests.post`, I just need to mock a `Request` object. I took inspiration from the above code [here](https://github.com/RasaHQ/rasa/blob/4d18c81fd159b93d157d40a4c709c4abfbb2cc64/tests/core/channels/test_slack.py#L93)

Reply:
No problem, thanks for looking into this 👍🏻